server/src/api_endpoints/clients/roles.js

```diff
@@ -13,11 +10,21 @@ module.exports = {
  * @param {*} res
  * @param {*} next
  */
-async function listClientRoles (req, res, next) {
-  const clientId = req.params.client_id
-
-  const roles = await dbAdminRole.listRoles(clientId)
-  res.status(200).send({ data: roles })
+const listClientRoles = async (req, res, next) => {
+  try {
+    const { client_id: clientId } = req.params
+    const { include_permissions } = req.query
+    await checkClientExists(clientId)
+    const roles = await dbAdminRoles.listRoles(clientId)
```

Comment:
@mikelax I have noticed that this API might return duplicate roles since it joins the tenant table. Additionally, the response schema is `{ data: [{}] }`, which is not consistent with our convention of `[{}, {}]`.

Making any changes to fix this could be considered a breaking change. I suggest we consider working on v3 to resolve all of these inconsistencies under that initiative.

Reply:
@k1ch  I looked over the query in the `listRoles` function and I see the join. The condition is called out in the documentation of the function. 

So I believe the same role being returned multiple times would only happen if a `clientId` is not supplied as a parameter to the function. 

Since this API is "client" specific and the database function will always be called with a clientId I think the logic is fine.

Reply:
@mikelax Check the API responses from both `GET /roles?client_id=the-usher` and `GET /clients/the-usher/roles` below. If a client is associated with multiple tenants, the APIs return duplicate roles due to a join with the tenant table.

![image](https://github.com/user-attachments/assets/d72e9b7b-0184-49d0-9a4b-9350ba1d723e)
![image](https://github.com/user-attachments/assets/7cf0da98-bfdf-4700-8b44-b853c8a4b57b)

Reply:
@mikelax I do not think it makes sense to include tenant-related metadata (`tenantname`, `iss_claim`) in the role object. Yes, I believe it should be a separate PR as part of v3 to fix this issue and update the schema, since removing fields and changing the response schema is a breaking change.